### PR TITLE
Add filter function for land overpasses.

### PR DESCRIPTION
**Pull request by JakobDeutloff**
Adds _filter_land() to radiometer level1 processing to filter out data taken over land.

### CODE REVIEW

File: orcestra/postprocess/level1.py [hunk 1]
```diff
@@ -148,6 +149,45 @@ def _filter_spikes(ds, threshold=5, window=1200):
...
+    Filters out data that was collected over land.
+    Removes data by offset earlier than the the time the plane flies over land
+    to avoid including land measurements due to tilt of the plane or mask inaccuracies.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset to filter.
+    mask : xr.DataArray
+        Mask of land and sea. 1 for sea, 0 for land.
```
Comment (lkluft) [inline orcestra/postprocess/level1.py:163]: How about naming this variable `sea_land_mask` (as you defined sea as 1)?

File: orcestra/postprocess/level1.py [hunk 2]
```diff
@@ -148,6 +149,45 @@ def _filter_spikes(ds, threshold=5, window=1200):
...
+    lat : xr.DataArray
+        Latitudes of the path.
+    lon : xr.DataArray
+        Longitudes of the path.
```
Comment (lkluft) [inline orcestra/postprocess/level1.py:167]: Would it be safe to assume that the input dataset is well-formatted and thus has a `lat` and `lon` variable? This way one could simplify the function signature a lot

File: orcestra/postprocess/level1.py [hunk 3]
```diff
@@ -148,6 +149,45 @@ def _filter_spikes(ds, threshold=5, window=1200):
...
+    mask_path_subsampled = mask_path.sel(time=ds.time, method="nearest").assign_coords(
+        time=ds.time
+    )
+    diff = (mask_path_subsampled * 1).diff("time")
+    start_land = diff.where(diff == -1).dropna("time").time
+    end_land = diff.where(diff == 1).dropna("time").time
+    for t in start_land:
+        mask_path_subsampled.loc[dict(time=slice(t - offset, t))] = 0
+    for t in end_land:
+        mask_path_subsampled.loc[dict(time=slice(t, t + offset))] = 0
```
Comment (lkluft) [inline orcestra/postprocess/level1.py:187]: Would it also be possible to just filter the, e.g. 30s running mean of the sea-land track?

**Review by lkluft — COMMENTED** [review]
Hi Jakob, thanks for working on this! 

In addition to some minor formatting comments, we might briefly talk about the function signature to keep the user interface lean

File: orcestra/postprocess/level1.py (hunk 1 above)
Comment (JakobDeutloff) [reply orcestra/postprocess/level1.py:163]: Sounds good, will implement that.

**Comment by lkluft** [comment]
This branch is now in conflict with changes in the restructuring of the post-processing. Are the changes here still relevant or already included in there?

**Pull request closed without merging**